### PR TITLE
RL-151 read data from new end point (Combined FE and BE)

### PR DESCRIPTION
**Pull request by amanbop**
### Jira Story
RL-151 - Read Data from Screening Relationship Endpoint (FE-Ruby)
https://osi-cwds.atlassian.net/browse/RL-151
RL-177 - Read Data from Screening Relationship Endpoint (FE-Redux React)
https://osi-cwds.atlassian.net/browse/RL-177

## Description
This is the combined PR for reading relationships for a screening from our new endpoint screenings/{Id}/relationships_with_candidates. This change will bring data from this new endpoint for screenings only. Snapshot will not be affected at all.

## Tests
- [x] I have included unit tests 
- [x] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 
This is a transparent change for Screenings. The old tests should work. I have added more unit tests to ensure correct endpoint is being called.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/controllers/api/v1/relationships_controller.rb [hunk 1]
```diff
@@ -30,6 +35,10 @@ def update
       def relationship_params
         params.require(:relationship).as_json.symbolize_keys
       end
+
+      def client_ids
+        params[:clientIds]&.split ','
```
Comment (money) [inline app/controllers/api/v1/relationships_controller.rb]: no need for `&`, just do `.split(',')

**Comment by amanbop** [comment]
retest this please

**Comment by wmitchell-cambria** [comment]
With the tests passing, I should clarify the remaining parts of my review:
1. Relationship/HOI fetching order in sagas - I don't really care too much about this, and I'm not willing to block the PR on it.
2. screeningId=null having different behavior than screeningId not being present - I do think this is a likely source of future errors and it will be difficult to debug in those situations, so I'd like to see it resolved before merging. That being said, if you have a strong reason for including it, I'm somewhat open to being convinced.

File: app/repositories/relationships_repository.rb [hunk 2]
```diff
@@ -14,6 +13,15 @@ def self.search(security_token, client_ids)
     ).body
   end
 
+  def self.get_relationships_for_screening_id(security_token, screeing_id)
+    return [] if screeing_id.blank?
```
[the remark this answers is not in the document]
Comment (amanbop) [reply app/repositories/relationships_repository.rb]: This has been fixed. The screeningId is truly optional now.

**Comment by amanbop** [comment]
screeningId=null having different behavior than screeningId not being present - This should be fixed now.

**Comment by wmitchell-cambria** [comment]
Awesome! I feel much safer having that in place.

**Pull request merged**